README.md
Introduction
------------
Proxy Haarpcache es un tenedor de Thundercache.

Lo resaltante en el sistema haarpcache, es que este permite almacenar los archivos en forma inteligente. 

Es decir almacena cada pedazo del video o archivo juntandolo todo en un único recurso en el disco, sin importar que estas partes hayan venido incompletas o en desorden. Además esto hace que 
cuando se pida de nuevo el mismo archivo; se despachen las partes que están en el disco y las que no se encuentran en el, se toman desde la red, pero al final las partes faltantes del archivo que se descargan se siguen almacenando en el mismo archivo; completando así el archivo original. 
 

Building
--------
* Type _./configure_ and then _make_
* To clean, type _make clean_


Changelog
---------
__Vesion 1.1__

2013/02/09

* Cache dinamico e inteligente.
* Plugins editados, para funcionamiento inteligente.
* Multi disco
* Bloqueo de publicidad en www.youtube.com.
* Cache de Youtube en html5.

Fork creado por

Kei Kurono

kei.haarpcache@gmail.com

http://perufw.wordpress.com


ChangeLog

*HAVP 0.91 released
 - Fix possible segfault on dns lookups (thanks Gavin McCullagh)
 - Fix compiling with gcc 4.4
 - Support AVG version 8.5 (default AVGPORT 54322) (thanks Markus Wigge)
   (note: you may need DISABLELOCKINGFOR AVG:ALL, which is now default)

*HAVP 0.90 released
 - ClamAV library 0.95 support (recompile needed)
 - Support NOD32 version 3 (set NOD32VERSION 30 in config)
 - Add PRELOADZIPHEADER config (Squid 3.x might not work if enabled)
 - Add SYSLOGVIRUSLEVEL config

*HAVP 0.89 released
 - Fix possible retry loop and hang (thanks to Peter Warasin @ endian.it)
 - Always send Via: header, fixes some IIS problems (e.g. MSNBC)

*HAVP 0.88 released
 - ClamAV library 0.93 support (recompile needed, new option CLAMMAXSCANSIZE)
 - CLAMMAXFILESIZE default is now 100MB (so 0.93 even starts scanning big files)
 - Fix random seed issue (ClamAV generated some temporary file errors)
 - Added DISABLELOCKINGFOR config (fix for ZIP handling in ClamAV 0.93)
 - Arcavir version 2008 support (set ARCAVIRVERSION)
 - Log scanner errors to errorlog
 - Relaxed SSL/CONNECT port limits
   (It is _not_ recommended to use --enable-ssl-tunnel, you should use Squid)

*HAVP 0.87 released
 - DrWeb scanner support
 - F-Prot support for v6.0 added (also check FPROTOPTIONS)
 - If false, X_FORWARDED_FOR drops also Via: header for privacy
 - Fix Avast and AVG bugs
 - Templates support <!--url--> and <!--clientip--> tags
 - Uses supplementary groups for user if defined
 - Added TRICKLINGBYTES config
 - Reduced *MAXFILES settings to 50 for performance
 - Add missing HTTP methods (MKACTIVITY, CHECKOUT, MERGE) 

*HAVP 0.86 released
 - Experimental support for chunked Transfer-Encoding, fixes some broken sites
 - Added IGNOREVIRUS config for whitelisting virus names
 - Added CLAMBLOCKBROKEN config
 - Improve detection with AVG, only Viruses were found, not Trojans etc
 - HAVP is killed if database reloading fails for Library Scanner
 - Log URL when crashed scanner process detected, for troubleshooting
 - Build system updated (--prefix --sbindir --sysconfdir --localstatedir)

*HAVP 0.85 released
 - Added support for ClamAV 0.90 library
   (once compiled for it, older library versions not supported)

*HAVP 0.84 released
 - Fix bug in tcp buffer, requests could leak to other clients sometimes
 - Support for Arcavir scanner
 - Support for multiple IPs in hostnames, all are tried if necessary
 - Pass Proxy-Authorization header to parent proxy (thanks Mateus)
 - Ignore scanner errors if MAXSCANSIZE reached (thanks Vittorio)
 - Build system polished a bit (--prefix supported)
 - Added support for old NOD32 scanner (set NOD32VERSION in config)
 - Default for MAXSCANSIZE 5000000, not suggested to be 0 anymore

*HAVP 0.83 released
 - Allow Accept-Encoding header so HTTP compression works
 - Workaround for some ClamAV mmap() troubles (affected BinHex and PDF)

*HAVP 0.82 released
 - Improved ZIP handling (header pre-fetch, scans ZIPs larger than MAXSCANSIZE)
 - Added SCANIMAGES config
 - Ignore RAR errors from clamd
 - Fixes to HTTP header handling
 - Added syslog logging option
 - Signal HUP re-opens logfiles, making rotation possible

*HAVP 0.81 released
 - Fixed binding to low ports (<1024)
 - Fixed FreeBSD, downloads that took longer than TRICKLING did not work

*HAVP 0.80 released
 - Experimental FreeBSD support (no mandatory locking, KEEPBACK not supported!)
 - Avast! scanner support
 - Added MAXDOWNLOADSIZE config
 - Added X_FORWARD_FOR config to control the header
 - Added some archive scanning parameters for Trophie
 - Added TCP support for clamd
 - Ignore RAR errors from ClamAV (use ClamAV-devel if you want to scan RARv3)
 - Fixed bug in socket buffer, sometimes caused nasty effects with POST etc.
 - Fixed KeepAlive for HTTP/1.1 clients, now on by default
 - Access logging format changed a bit

*HAVP 0.79 released
 - MULTIPLE SCANNER SUPPORT! (see havp.config how to enable scanners)
 - NOD32, Sophos and Clamd scanner support
 - Parentproxy error on SSL tunneling is passed to browser
 - Whitelisted sites can use HTTP Range requests (for Windowsupdate..)
 - Added STREAMUSERAGENT/STREAMSCANSIZE config to reduce stream scanning
 - Added SCANNERTIMEOUT option to catch scanners gone wild
 - Added scanning options for ClamLib
 - Some HTTP header handling fixes
 - Init-script updated so it works on Solaris

*HAVP 0.78 released
 - Solaris is now supported too
 - AVG scanner support
 - Kaspersky aveserver socket support (faster)
 - Added TEMPDIR, LOGLEVEL, FAILSCANERROR and WHITELISTFIRST config
 - KEEPBACKTIME config added to complement KEEPBACKBUFFER setting
 - Basic HTTP Keep-Alive support, improves network performance
 - HTTPS/SSL tunneling support (requires configure --enable-ssl-tunnel)
 - FTP is supported and scanned when FTP supporting parent proxy is used
 - Default user and group changed to havp
 - Logging improved
 - Many important and not so important fixes/optimizations

*HAVP 0.77 released
 - Fix Security Bug - Kaspersky and F-Prot did not catch viruses
 - Add Trophie (Trend Micro) Scanner
 - Added RANGE config for improved security
 - Change some default settings

*HAVP 0.76 released
 - Improve performance
 - Fix small bug in logfile 
 - Swedish Templates <Johnny Strom>
 - Polish Templates <Eryk L. Rathmann>

*HAVP 0.75 - 02/01/2005
 - Improve Configfile
 - Fix Bug in Black/Whitelists

*HAVP 0.74 - 12/11/2005
 - Add IP to Logfile
 - Add Templates
 - Improve Errorhandling
 - Fix Bug in Serverpool

*HAVP 0.73 - 20/10/2005
 - Move Pidfile to /var/run/havp/havp.pid
 - Add Templates
 - Improve Performance
 - Rewrite Domains (Check default.h.in)

*HAVP 0.72 - 07/10/2005
 - Add Kaspersky and F-Prot Scanner
 - Add Templates
 - Fix some minor bugs in default.h



haarp/default.h
/* haarp/default.h.  Generated from default.h.in by configure.  */
#ifndef DEFAULT_H
#define DEFAULT_H

#define VERSION "1.1"

//##############################################################
//Parameters in Configurationfile

#define CONFIGPARAMS \
 "USER","GROUP", \
 "SERVERNUMBER","PORT","BIND_ADDRESS","SOURCE_ADDRESS", "PIDFILE", \
 "DAEMON","TRANSPARENT","LOG_OKS","ACCESSLOG","ERRORLOG","LOGLEVEL", \
 "USESYSLOG","SYSLOGNAME","SYSLOGFACILITY","SYSLOGLEVEL", "DISPLAYINITIALMESSAGES", \
 "PARENTPROXY","PARENTPORT","MAXSERVERS","FORWARDED_IP","X_FORWARDED_FOR", \
 "CACHEDIR","PLUGINSDIR","MYSQL_HOST","MYSQL_USER","MYSQL_PASS","MYSQL_DB","CACHE_LIMIT"


//##############################################################
//Configuration not setable in havp.config

//Time format
#define TIMEFORMAT "%d/%m/%Y %H:%M:%S "

//CONNTIMEOUT in seconds
#define CONNTIMEOUT 60

//RECVTIMEOUT in seconds
#define RECVTIMEOUT 120

//SENDTIMEOUT in seconds
#define SENDTIMEOUT 120

//Maximum client connection waiting for accept
#define MAXCONNECTIONS 1024

//Maximum bytes received in one request
#define MAXRECV 14600

//Maximum logfile line length
#define STRINGLENGTH 1000

//Valid Methods
#define METHODS \
 "GET","POST","HEAD","CONNECT","PUT","TRACE","PURGE","OPTIONS","UNLOCK", \
 "SEARCH","PROPFIND","BPROPFIND","PROPPATCH","BPROPPATCH","MKCOL","COPY", \
 "BCOPY","MOVE","LOCK","BMOVE","DELETE","BDELETE","SUBSCRIBE","UNSUBSCRIBE", \
 "POLL","REPORT","ERROR","NONE","MKACTIVITY","CHECKOUT","MERGE"

//Maximum length of http headers
#define MAXHTTPHEADERLENGTH 65536

// DONT TOUCH - run configure
#define CONFIGFILE "/etc/haarp/haarp.conf"
#define ACCESSLOG "/var/log/haarp/access.log"
#define ERRORLOG "/var/log/haarp/error.log"
#define PIDFILE "/var/run/haarp/haarp.pid"
/* #undef SSLTUNNEL */
#define HAVE_SETGROUPS 1
#define HAVE_INITGROUPS 1

#endif
